gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -1020,6 +1035,32 @@ public SubscriptionEntity resumeConsumer(final ExecutionContext executionContext
         }
     }
 
+    private SubscriptionEntity updateSubscription(ExecutionContext executionContext, Subscription subscription, String apiId)
+        throws TechnicalException {
+        Subscription previousSubscription = new Subscription(subscription);
+        final Date now = new Date();
+        subscription.setUpdatedAt(now);
+        subscription.setConsumerPausedAt(null);
+        subscription.setConsumerStatus(Subscription.ConsumerStatus.STARTED);
+
+        subscription = subscriptionRepository.update(subscription);
+
+        createAudit(
+            executionContext,
+            apiId,
+            subscription.getApplication(),
+            SUBSCRIPTION_RESUMED_BY_CONSUMER,
+            subscription.getUpdatedAt(),
+            previousSubscription,
+            subscription
+        );
+
+        // active API Keys are automatically unpause
+        resumeApiKeys(executionContext, subscription);
```

Comment:
I don't like having this here in a "updateSubscription" -> resumeApiKeys is about resuming something, update is a too generic term.
why not calling the method `resumeSubscription`

Reply:
The same decision as below to treat it like a seperate case

Reply:
sorry, i don't understand the answer :)

Reply:
Ok, so I will also ask about question. There is no such method as `resumeSubscription` in `SubscriptionServiceImpl`

Reply:
yes... that is why I propose to rename `updateSubscription` to `resumeSubscription`, because `updateSubscription` is not correct as it does some "resume" action, by calling `resumeApiKeys()`.
the risk with `updateSubscription` name is that it can be call by a classic update method and it will unpause the apiKeys, and it is a side effect we might not want. does it make sense ?